src/pkg/services/m365/api/drive_test.go

```diff
@@ -51,3 +72,195 @@ func (suite *OneDriveAPISuite) TestCreatePagerAndGetPage() {
+				assert.Nil(t, i)
+			},
+		},
+		{
+			name:        "rename folder",
+			onCollision: control.Copy,
+			postItem:    folder,
+			expectErr: func(t *testing.T, err error) {
+				assert.NoError(t, err, clues.ToCore(err))
+			},
+			expectItem: func(t *testing.T, i models.DriveItemable) {
+				assert.NotEqual(
+					t,
+					ptr.Val(origFolder.GetId()),
+					ptr.Val(i.GetId()),
+					"replaced item should be different from the original")
+			},
+		},
+		{
+			name:        "replace folder",
+			onCollision: control.Replace,
+			postItem:    folder,
+			expectErr: func(t *testing.T, err error) {
+				assert.NoError(t, err, clues.ToCore(err))
+			},
+			expectItem: func(t *testing.T, i models.DriveItemable) {
+				assert.Equal(
+					t,
+					ptr.Val(origFolder.GetId()),
+					ptr.Val(i.GetId()),
+					"replaced item should be the same as the original")
+			},
+		},
+		{
+			name:        "fail file",
+			onCollision: control.Skip,
+			postItem:    file,
+			expectErr: func(t *testing.T, err error) {
+				assert.ErrorIs(t, err, graph.ErrItemAlreadyExistsConflict, clues.ToCore(err))
+			},
+			expectItem: func(t *testing.T, i models.DriveItemable) {
+				assert.Nil(t, i)
+			},
+		},
+		{
+			name:        "rename file",
+			onCollision: control.Copy,
+			postItem:    file,
+			expectErr: func(t *testing.T, err error) {
+				assert.NoError(t, err, clues.ToCore(err))
+			},
+			expectItem: func(t *testing.T, i models.DriveItemable) {
+				assert.NotEqual(
+					t,
+					ptr.Val(origFile.GetId()),
+					ptr.Val(i.GetId()),
+					"replaced item should be different from the original")
+			},
+		},
+		// FIXME: this *should* behave the same as folder collision, but there's either a
+		// bug or a deviation in graph api behavior.
+		// See open ticket: https://github.com/OneDrive/onedrive-api-docs/issues/1702
+		{
+			name:        "replace file",
+			onCollision: control.Replace,
+			postItem:    file,
+			expectErr: func(t *testing.T, err error) {
+				assert.ErrorIs(t, err, graph.ErrItemAlreadyExistsConflict, clues.ToCore(err))
+			},
+			expectItem: func(t *testing.T, i models.DriveItemable) {
+				assert.Nil(t, i)
+			},
+		},
+	}
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+			i, err := suite.ac.Drives().PostItemInContainer(
+				ctx,
+				suite.driveID,
+				ptr.Val(parent.GetId()),
+				test.postItem,
+				test.onCollision)
+
+			test.expectErr(t, err)
+			test.expectItem(t, i)
+
+			if err != nil {
+				return
+			}
+
+			if test.postItem == folder {
+				return // no uploadSession needed
+			}
+
+			mus, err := suite.ac.Drives().NewItemContentUpload(
+				ctx,
+				suite.driveID,
+				ptr.Val(i.GetId()))
+			require.NoError(t, err, clues.ToCore(err))
+
+			iw := graph.NewLargeItemWriter(
+				ptr.Val(i.GetId()),
+				ptr.Val(mus.GetUploadUrl()),
+				int64(len(test.name)))
+
+			_, err = iw.Write([]byte(test.name))
+			require.NoError(t, err, clues.ToCore(err))
```

Comment:
is there a particular reason to upload data here? It's not really checked (i.e. don't ensure the new file's content changed rather than the old) so it doesn't seem to add much to the test. There's also other tests that check uploading data (albeit more indirectly)

Reply:
Oh!  That's leftover from bug investigation.  Thanks for the reminder.